From def0618d059eadfb4704d2f9b1f7a5379d89fdaf Mon Sep 17 00:00:00 2001
From: Ying Xue <ying.xue@windriver.com>
Date: Thu, 29 Nov 2012 16:28:30 +0800
Subject: [PATCH 181/183] tipc: eliminate connection setup for implied connect
 in recv_msg()
Git-commit: def0618d059eadfb4704d2f9b1f7a5379d89fdaf
Patch-mainline: Merged into tipc-devel
References: bnc#797455

As connection setup is now completed asynchronously in BH context,
in the function filter_connect(), the corresponding code in recv_msg()
becomes redundant.

Signed-off-by: Ying Xue <ying.xue@windriver.com>
Signed-off-by: Jon Maloy <jon.maloy@ericsson.com>
Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
Signed-off-by: Jiri Kosina <jkosina@suse.cz>

---
 net/tipc/socket.c |    7 -------
 1 files changed, 0 insertions(+), 7 deletions(-)

diff --git a/net/tipc/socket.c b/net/tipc/socket.c
index a724ecc..cd178fe 100644
--- a/net/tipc/socket.c
+++ b/net/tipc/socket.c
@@ -945,13 +945,6 @@ restart:
 	sz = msg_data_sz(msg);
 	err = msg_errcode(msg);
 
-	/* Complete connection setup for an implied connect */
-	if (unlikely(sock->state == SS_CONNECTING)) {
-		res = auto_connect(sock, msg);
-		if (res)
-			goto exit;
-	}
-
 	/* Discard an empty non-errored message & try again */
 	if ((!sz) && (!err)) {
 		advance_rx_queue(sk);
-- 
1.7.8.3

